static/js/components/input/ProductSelector.js

```diff
@@ -180,8 +215,12 @@ export default class ProductSelector extends React.Component<Props, State> {
     }
 
     if (productType === PRODUCT_TYPE_PROGRAM) {
-      // $FlowFixMe: Can't be null/undefined
-      productValue = selectedCoursewareObj.value
+      // This is a dirty hack to support program run tags. Refer to `B2bPurchasePage.onSubmit` for further info.
```

Comment:
We're seeing the limitations of treating this set of multiple selects as a single input. Still, I think we can keep this approach and just set the `productValue` as an object in all cases. 

```javascript
if (productType === PRODUCT_TYPE_PROGRAM) {
  productValue = {
    // $FlowFixMe: Can't be null/undefined 
    productId:  selectedCoursewareObj.value,
    programRunId: selectedCourseDate ? selectedCourseDate.value : null
  }
} else {
  productValue = {
    // $FlowFixMe: Can't be null/undefined 
    productId: selectedCourseDate.value,
    programRunId: null
  }
}
```

In that case I think you can amend this comment just to say that this input is keeping track of multiple values, and therefore the value is an object. You can remove the comment in `B2BPurchasePage.js`

Reply:
Updated